File: evals/act/bidnet.ts
import { initStagehand } from "../utils";
import { EvalFunction } from "../../types/evals";

export const bidnet: EvalFunction = async ({ modelName, logger }) => {
  const { stagehand, initResponse } = await initStagehand({
    modelName,
    logger,
  });

  const { debugUrl, sessionUrl } = initResponse;

  await stagehand.page.goto("https://www.bidnetdirect.com/");

  await stagehand.act({
    action: 'Click on the "Construction" keyword',
  });

  const expectedUrl =
    "https://www.bidnetdirect.com/public/solicitations/open?keywords=Construction";
  const currentUrl = stagehand.page.url();

  await stagehand.close();

  return {
    _success: currentUrl.startsWith(expectedUrl),
    currentUrl,
    debugUrl,
    sessionUrl,
    logs: logger.getLogs(),
  };
};

File: packages/sdk-ruby/evals/tasks/act/bidnet.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/bidnet.ts

Evals.define_task("bidnet") do |t|
  t.page.goto("https://www.bidnetdirect.com/")

  t.stagehand.act('Click on the "Construction" keyword')

  expected_url = "https://www.bidnetdirect.com/public/solicitations/open?keywords=Construction"
  current_url = t.page.url

  { _success: current_url.start_with?(expected_url), currentUrl: current_url }
end
